### PR TITLE
Add --label / -l which prefixes lines with text

### PR DESCRIPTION
**Pull request by DanielO**
I am using this to add a label to the output when decoding multiple channels.

### CODE REVIEW

File: unixinput.c [hunk 1]
```diff
@@ -582,7 +586,8 @@ static const char usage_str[] = "\n"
         "  -g         : CW: Gap length in ms (default: 50)\n"
         "  -x         : CW: Disable auto threshold detection\n"
         "  -y         : CW: Disable auto timing detection\n"
-		"  --timestamp: Add a time stamp in front of every printed line\n"
+        "  --timestamp: Add a time stamp in front of every printed line\n"
+        " --label     : Add a label to the front of every printed line\n"
```
Comment (davidmckenzie) [inline unixinput.c]: Missing a space at the beginning of the line for matching indentation :)

**Comment by DanielO** [comment]
Oops, fixed. That's what I get for committing while drinking.. ;)

**Comment by EliasOenal** [comment]
I am overall fine with the feature, but I would prefer to avoid strdup() since it isn't standard C. In addition the buffer is never freed which may result in a memory leak on some systems. Lastly I think strdup may be omitted altogether since optarg should just be a pointer to argv.

**Comment by DanielO** [comment]
I believe strdup is part of POSIX - http://pubs.opengroup.org/onlinepubs/009695399/functions/strdup.html
At a practical level I believe it's available everywhere (gcc, clang, MSVC++, etc)

That said I think you are right WRT argv, https://stackoverflow.com/questions/963493/is-it-possible-to-change-argv-or-do-i-need-to-create-an-adjusted-copy-of-it shows C99 saying it doesn't change.

I didn't bother freeing memory because exit will do that ;)

**Comment by EliasOenal** [comment]
I do a lot of embedded development and thus my views may be a bit skewed. But overall I would prefer avoiding POSIX in favour of portability. Also memory being freed on exit is just a feature of many modern operating systems, on some platforms this may not be available.

**Comment by EliasOenal** [comment]
I just noticed --label only works when --timestamp is used as well. Could you please fix this?

**Comment by DanielO** [comment]
Done

**Comment by EliasOenal** [comment]
Please open a pull-request for the fix.

**Pull request merged**